### DISCUSSION TITLE
reTerminal DM Flash OS | Seeed Studio Wiki

### DISCUSSION THREAD

**Discussion by giscus** [Seeed-Studio/wiki-documents#555] [Q&A] [answered]
# reTerminal DM Flash OS | Seeed Studio Wiki

reTerminal DM Flash OS

https://wiki.seeedstudio.com/reterminal-dm-flash-OS/

**Comment by iGridAU** [comment]
Hi - after fixing the touch screen orientation, touch input is not not rotated to match the display. Can you assist? Thanks!

**Reply by iGridAU** [reply]
touch screen input not matching screen rotation is only broken on Raspberry Pi OS 32 bit. 64 bit works without issue.

**Reply by xzxcessarr** [reply, marked as answer]
Hello, thank you for feedback! here is the solution we provide for you:
1. reboot you system and using the ssh to enter the address of " /etc/X11/xorg.conf.d/", and make a file named 50-goodix-map.conf
2. copy and paste the content below(we will offer you a screenshot):
Section "InputClass"
    Identifier "Goodix Capacitive TouchScreen Setting"
    MatchProduct "Goodix Capacitive TouchScreen"
    Option "CalibrationMatrix" "1.000000 0.000000 0.000000 0.000000 1.000000 0.000000 0.000000 0.000000 1.000000"
EndSection
3. save and check the file is all right by: vim /etc/X11/xorg.conf.d/50-goodix-map.conf
4. open the terminal and paste the query: systemctl restart lightdm
5. reboot and check if the screen and touch
The screenshot is here: https://imgse.com/i/pP3kzs1

**Comment by Keith-Bravo** [comment]
Note: After you add " echo arm_64bit=0 | sudo tee -a /boot/config.txt" you then need to reboot and then proceed with installing the drivers.

**Comment by hayderismael** [comment]
I have installed the OS on reTerminal DM and followed all steps of installation and drivers updates. The LCD works well and touch orientation, too. If I connect an external mouse via a USB port, that works ok. The only problem when I remotely access the pi via VNC the mouse orientation via the VNC is 90 deg oriented and so difficult to go anywhere. Please help.

**Comment by leematos** [comment]
Adding a note here -- I was able to follow https://wiki.seeedstudio.com/reTerminal-FAQ/#q12-troubleshooting-touch-screen-inaccuracy after installing raspbian 64 bit and this fixed my touch screen issues. 👍

**Comment by cgjgh** [comment]
Are there any instructions for booting with NVME SSD? Have cloned os from eMMC, updated bootloader to latest version, set BOOT_ORDER=0xf21546  (Try NVMe PCIe first, followed by, USB PCIe, USB SoC XHCI, SD/eMMC then network) and it says failed to open device 'nvme', and proceeds to boot from eMMC. The drive I'm using is ([Seeed Studio SSD](https://www.seeedstudio.com/M-2-2280-SSD-128GB-p-5332.html)).

Much appreciated.

**Comment by LJ-Hao** [comment]
Could you update your eeprom follow this [link](https://wiki.seeedstudio.com/raspberry_pi_5_uses_pcie_hat_dual_hat/#use-sd-card-copier-tool-to-flash-os-onto-the-nvme-ssd-on-raspberry-pi-os), and then try reboot again.

**Reply by cgjgh** [reply]
Are these instructions specific to RPi 5, as the reTerminal DM uses RPi Compute Module 4?

**Reply by LJ-Hao** [reply]
It will work on both RPI5 and RPI Compute Module 4.

**Comment by mpfj** [comment]
Just installing a 32bit OS and after the line

`echo arm_64bit=0 | sudo tee -a /boot/config.txt`

I had to do `sudo reboot` before installing the drivers.
Otherwise the driver install failed.

Not sure if that's specific to me but, if not, might be worth updating the Wiki.

**Reply by nfs0619** [reply]
Thank you for your feedback! We will arrange to test the situation you mentioned, and if the issue is confirmed, we will update the documentation accordingly. Thanks again for your support!

**Reply by Kian-kang** [reply]
Thank you for your feedback.After conducting tests,we have indeed identified this issue.We will update the WiKi documentation

**Comment by Bharath567390** [comment]
Hello, I have uploaded the raspberry pi os with 32bit but the screen is blank . Provide me the solution.

**Reply by mpfj** [reply]
This is normal.
By default, the display drivers, etc are not part of the normal RPi OS so you now need to install the custom Seeed Studio drivers.
Follow the instructions from https://wiki.seeedstudio.com/reterminal-dm-flash-OS/#install-drivers onwards.

**Reply by ReKST-117** [reply]
Hi mpfj, Thanks for your reply:) 

Hi Bharath567390, yes, you need to follow the Wiki how to install the driver. Thanks!

**Comment by Bharath567390** [comment]
Thank you

**Comment by Bharath567390** [comment]
Hello sir,
I want to link the url page into my hmi display. Is there any possible ways for url link in kiosk mode of raspberry pi cm4 with integrated 10 inch hmi display for external purpose

**Reply by ReKST-117** [reply]
Hi Bharath, it is possible to display an external webpage on your Raspberry Pi CM4 HMI display in kiosk mode. The simplest way is to use the Chromium browser’s kiosk option, which automatically opens a specified URL in full screen after boot. This setup is commonly used for dashboards and HMI interfaces, but please note that this suggestion is provided for reference only. Actual performance may depend on your operating system and configuration.

**Comment by mpfj** [comment]
The latest Raspbian images DO NOT work on this unit. The seeed drivers fail to install.
I have raised a bug here : https://github.com/Seeed-Studio/seeed-linux-dtoverlays/issues/142

Until this is fixed, you need to use Raspbian Bullseye which you can download from https://downloads.raspberrypi.com/raspios_oldstable_armhf/images/raspios_oldstable_armhf-2025-05-07/

**Reply by mpfj** [reply]
Seems this might relate to the 32bit OS rather than the 64bit version.
For various reasons, I need to use the 32bit version so I may have to stick with Bullseye.
Unless anyone can guide me to a solution?
